File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/CreatorProfile.java
package com.eager.questioncloud.core.domain.creator;

import com.eager.questioncloud.core.domain.question.Subject;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class CreatorProfile {
    private Subject mainSubject;
    private String introduction;

    public static CreatorProfile create(Subject mainSubject, String introduction) {
        return new CreatorProfile(mainSubject, introduction);
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/creator/dto/CreatorProfile.kt
package com.eager.questioncloud.core.domain.creator.dto

import com.eager.questioncloud.core.domain.question.enums.Subject

class CreatorProfile(
    val creatorId: Long,
    val name: String,
    val profileImage: String?,
    val mainSubject: Subject,
    val email: String,
    val introduction: String,
) {
}